pipeline-model-definition/src/main/resources/org/jenkinsci/plugins/pipeline/modeldefinition/ModelInterpreter.groovy

```diff
@@ -247,9 +264,26 @@ public class ModelInterpreter implements Serializable {
         }
     }
 
+    private Map<String,CredentialWrapper> processCredentials(@Nonnull List<List<String>> varsAndIds) {
+        Map<String,CredentialWrapper> creds = new TreeMap<>()
+        RunWrapper currentBuild = script.getProperty("currentBuild")
+
+        for (int i = 0; i < varsAndIds.size(); i++) {
+            String key = varsAndIds.get(i)?.get(0)
+            if (key != null) {
+                String id = (String)script.evaluate(Utils.prepareForEvalToString(varsAndIds.get(i)?.get(1)))
```

Comment:
Oh, hey, I actually did the sandboxing elsewhere. Here is not actually a problem - `CpsScript.evaluate(script)` actually goes through `CpsScript.$getShell().evaluate(script)`, and that uses the `CpsThread.current().getExecution().getShell()`, which is not trusted, so will fail with `RejectedAccessException`s when appropriate. Phew! I did get rid of the `GroovyShell` we were using directly in `Environment` in favor of wrapping around `SecureGroovyScript.evaluate(classloader, binding)` so that area is sandboxed as well.